cmd/lynx-controller/main.go

```diff
@@ -121,3 +133,75 @@ func main() {
 		klog.Fatalf("error while running manager: %s", err.Error())
 	}
 }
+
+func setWebhookCert(k8sReader client.Reader) {
+	k8sClient := k8sReader.(client.Client)
+
+	// get secret
+	secretKey := types.NamespacedName{
+		Name:      "lynx-controller-tls",
+		Namespace: "kube-system",
+	}
+	secret := corev1.Secret{}
+	if err := k8sClient.Get(context.Background(), secretKey, &secret); err == nil {
+		// secret has been updated
+		if len(secret.Data["tls.crt"]) != 0 {
+			return
+		}
+
+		// create ca & caKey
+		caConf := &cert.CertConfig{
+			Config: certutil.Config{
+				CommonName:   "everoute",
+				Organization: []string{"SmartX"},
+				Usages:       []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
+			},
+			PublicKeyAlgorithm: x509.RSA,
+		}
+		ca, caKey, _ := cert.NewCertificateAuthority(caConf)
+
+		// sign a new tls cert
+		tlsConf := &cert.CertConfig{
+			Config: certutil.Config{
+				CommonName:   "everoute",
+				Organization: []string{"SmartX"},
+				AltNames: certutil.AltNames{
+					DNSNames: []string{"lynx-validator-webhook.kube-system.svc"},
+					IPs:      []net.IP{net.ParseIP("127.0.0.1")},
+				},
+				Usages: []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
+			},
+			PublicKeyAlgorithm: x509.RSA,
+		}
+		tls, tlsKey, _ := cert.NewCertAndKey(ca, caKey, tlsConf, time.Now().AddDate(100, 0, 0))
+		tlsKeyByte, _ := keyutil.MarshalPrivateKeyToPEM(tlsKey)
+
+		// update secret
+		secret.Data["tls.crt"] = append(secret.Data["tls.crt"], cert.EncodeCertPEM(tls)...)
+		secret.Data["tls.key"] = append(secret.Data["tls.key"], tlsKeyByte...)
+		if err = k8sClient.Update(context.Background(), &secret); err != nil {
+			return
```

Comment:
Add some log information about the update secret failure.

Reply:
Also add fatal log, only ONE controller could update secret successfully. Other controller will get correct secret after restarting.